### ISSUE TITLE
Add eac3to support

### ISSUE THREAD

**Issue by wieslawsoltes** [wieslawsoltes/BatchEncoder#27]

**Comment by Metaphorce812** [comment]
Adding support for eac3to along with the ability to trans-code audio streams within video files (#63) would make BatchEncoder the complete audio conversion tool I've been searching for for what feels like forever. eac3to has so many functions both as a decoder and encoder ([especially if you install the additional decoders/encoders](https://en.wikibooks.org/wiki/Eac3to/How_to_Use)), that it would be awesome if you could integrate it into BatchEncoder.

I tried to manually integrate it into BatchEncoder by doing the following:
Configure Formats -> Add Format
Under "Input Extensions" I added RAW,LPCM,PCM,WAV,W64,RF64,AC3,EAC3,DTS,DTSES,DTS9624,DTSHDHR,DTSHDMA,MLP,THD,THDAC3,FLAC,AAC
Under "Type" I selected "Encoder"
Under "Command-Line Template" I'm pretty sure I entered the correct syntax:
$EXE $INFILE $OUTFILE $OPTIONS
since [the command-line syntax listed on the eac3to wiki](https://en.wikibooks.org/wiki/Eac3to/How_to_Use#Command_Line_Syntax) is:
eac3to input.file output.file -options

Under "Progress Function" I originally tried "- none -" since I have no idea how to write a progress function for it.

Here's that code from the BatchEncoder.formats file:
```XML
<Format id="EAC3TO_AC3" name="AC3 - eac3to" template="$EXE $INFILE $OUTFILE $OPTIONS" input="true" output="false" function="- none -" path="D:\My Documents\Dropbox\Installers\!Portable Programs\!EAC3to + Tools &amp; GUIs\eac3to v3.31 (2015.11.14)\eac3to.exe" success="0" type="0" formats="RAW,LPCM,PCM,WAV,W64,RF64,AC3,EAC3,DTS,DTSES,DTS9624,DTSHDHR,DTSHDMA,MLP,THD,THDAC3,FLAC,AAC" extension="AC3" default="0">
        <Presets>
            <Preset name="640 kbps" options="-640"/>
            <Preset name="Extract AC3 core from TrueHD" options="-core"/>
            <Preset name="Downmix 7.1/6.1 to 5.1 640 kbps" options="-down6 -640"/>
        </Presets>
    </Format>
```
As you can see, I added a few presets (using what I'm fairly certain is the correct syntax) and tried to convert some audio files using the 640 kbps preset. 
When I had the "Pipes" checkbox for "Input" checked (like it is for all the other encoders), I immediately got the error "progress did not reach 100%"
When I unchecked "Input," I immediately got the error "can not load progress script"
So I tried to make a place-holder progress script (using the simplest existing one I could find) with the code `function GetProgress(s) 
  return "-1";
end`, named the file "GetProgress_Eac3toEnc.progress" and linked to it under "Progress function:" in the "Configure Formats" menu.
**This time when I tried to encode (with the "Input" checkbox under "Pipes" unchecked), the Status column actually showed "Encoding" for about how long I would expect the conversion to take, but then I got the error "progress did not reach 100%."**
When I tried the same thing but with the "Input" checkbox under "Pipes" checked, I got the same error, but immediately and without "Encoding" showing up under Status.

So I can't be sure that eac3to would work if I had a functional progress script, but it does seem pretty clear that that's the immediate problem preventing it from being successful. There is [an eac3to command called "-progressnumbers"](https://en.wikibooks.org/wiki/Eac3to/How_to_Use#Using_audio_or_video_files_as_input) with the description "outputs one new line to the command window for every progress (for GUIs)." I have no idea how to write progress scripts, but maybe with the info I've given you, you could figure it out. At least that's my hope. Considering one of my attempts ran for the amount of time I would expect the conversion to take before the "progress did not reach 100%" error, I think if you could make a working progress script, then you'd be able to integrate eac3to into BatchEncoder.

Here's the Format code that actually managed to show "Encoding" in the Status column before the progress error:
```XML
<Format id="EAC3TO_AC3" name="AC3 - eac3to (not working b/c of GetProgress?)" template="$EXE $INFILE $OUTFILE $OPTIONS" input="false" output="false" function="C:\Program Files\BatchEncoder v1.3 (64-bit) (Portable)\GetProgress_Eac3toEnc.progress" path="D:\My Documents\Dropbox\Installers\!Portable Programs\!EAC3to + Tools &amp; GUIs\eac3to v3.31 (2015.11.14)\eac3to.exe" success="0" type="0" formats="RAW,LPCM,PCM,WAV,W64,RF64,AC3,EAC3,DTS,DTSES,DTS9624,DTSHDHR,DTSHDMA,MLP,THD,THDAC3,FLAC,AAC" extension="AC3" default="0">
        <Presets>
            <Preset name="640 kbps" options="-640"/>
            <Preset name="Extract AC3 core from TrueHD" options="-core"/>
            <Preset name="Downmix 7.1/6.1 to 5.1 640 kbps" options="-down6 -640"/>
        </Presets>
    </Format>
```
When googling for "eac3to + pipes" I found [the developer's main post about eac3to](https://forum.doom9.org/showthread.php?t=125966) (which might be helpful), and at the bottom of the post he talks about and gives examples of using eac3to to pipe PCM to other encoders.

I hope this helps. I appreciate all your hard work on this program. Keep it up!

**Comment by wieslawsoltes** [comment]
@Metaphorce812 I have finally added support for eac3to and will be released as part of v2.0. You can download latest CI build for testing.

* Progress function:
```Lua
function GetProgress(s) 
  if string.match(s, '^Done.%s-') ~= nil then return "100";
  else return string.match(s, 'process: (%d+)%%'); end;
end
```

* Format AC3
```XML
<?xml version="1.0" encoding="UTF-8"?>
<Format id="EAC3TO_AC3" name="AC3 (ATSC A/52) - eac3to" template="$EXE $INFILE $OUTFILE -progressnumbers $OPTIONS" input="false" output="false" function="GetProgress_Eac3toEnc.progress" path="eac3to/eac3to.exe" success="0" type="0" formats="RAW,LPCM,PCM,WAV,W64,RF64,AC3,EAC3,DTS,DTSES,DTS9624,DTSHDHR,DTSHDMA,MLP,THD,THDAC3,FLAC,AAC" extension="AC3" default="3">
    <Presets>
        <Preset name="192 kbit/s" options="-192"/>
        <Preset name="224 kbit/s" options="-224"/>
        <Preset name="384 kbit/s" options="-384"/>
        <Preset name="448 kbit/s" options="-448"/>
        <Preset name="640 kbit/s" options="-640"/>
        <Preset name="Extract the AC3 core of a TrueHD track" options="-core"/>
    </Presets>
</Format>
```

* Format DTS
```XML
<?xml version="1.0" encoding="UTF-8"?>
<Format id="EAC3TO_DTS" name="DTS (Coherent Acoustics codec) - eac3to" template="$EXE $INFILE $OUTFILE -progressnumbers $OPTIONS" input="false" output="false" function="GetProgress_Eac3toEnc.progress" path="eac3to/eac3to.exe" success="0" type="0" formats="RAW,LPCM,PCM,WAV,W64,RF64,AC3,EAC3,DTS,DTSES,DTS9624,DTSHDHR,DTSHDMA,MLP,THD,THDAC3,FLAC,AAC" extension="DTS" default="0">
    <Presets>
        <Preset name="768 kbit/s" options="-768"/>
        <Preset name="1536 kbit/s" options="-1536"/>
        <Preset name="Extract the DTS core of a DTS-HD track" options="-core"/>
    </Presets>
</Format>
```

**Comment by Metaphorce812** [comment]
Great - thanks! I haven't had a chance to test it out yet, but I'll let you know if I run into any problems.
